content/docs/guides/neon-identity.md

```diff
@@ -0,0 +1,149 @@
+
+## Authentication and synchronization
+
+When implementing user authentication, it is common to use dedicated authentication providers like StackAuth, Clerk, and others to handle the complexities of user identity, passwords, and security. However, keeping your database in sync with these providers typically requires additional development work.
+
+Neon Identity solves this by maintaining a direct connection between your auth provider and database, ensuring your application always has access to up-to-date user information right from your database.
+
+_<b>Q:</b> What specific benefits can we list out here? Less API calls, etc?_
+
+## How Neon Identity works
+
+When you set up Neon Identity, we create a `neon_identity` schema in your database. As users authenticate and manage their profiles in your auth provider, their data is automatically synchronized to your database.
+
+Here is the basic flow:
+
+1. **User profiles are created and managed in your authentication provider**
+
+   This view shows the list of users inside your auth provider (e.g. StackAuth). When new users sign up or update their profiles in your app, their data first appears here:
+   ![Users in Auth Provider](/docs/guides/stackauth_users.png)
+
+2. **Neon Identity syncs their data to your database**
+
+   This view shows the synced user profiles in Neon Identity. This is where Neon manages the connection between your database and the authentication provider.
+   ![Same users in Neon Identity](/docs/guides/identity_users.png)
+
+3. **The data is immediately available in your database**
+
+   The synchronized data is immediately available in the `neon_identity.users_sync` table. Here's an example query to inspect the synchronized data:
+
+   ```sql
+   SELECT * FROM neon_identity.users_sync;
+   ```
+
+   | id          | name       | email             | created_at    | raw_json                      |
+   | ----------- | ---------- | ----------------- | ------------- | ----------------------------- |
+   | 21373f88... | Sarah Chen | sarah@acme.dev    | 2024-12-17... | \{"id": "21373f88-...", ...\} |
+   | 0310a9a5... | Alex Kumar | alex@startmeup.co | 2024-12-17... | \{"id": "0310a9a5-...", ...\} |
+
+### Table structure
+
+The following columns are included in the `neon_identity.users_sync` table:
+
+- `raw_json`: Complete user profile as JSON
+- `id`: The unique ID of the user
+- `name`: The user's display name
+- `email`: The user's primary email
+- `created_at`: When the user signed up
+
+Updates to user profiles in the auth provider are automatically synchronized.
+
+## Before and after Neon Identity
+
+Let's take a look at how Neon Identiy can help simplify the code in a typical todos application:
+
+### Before Neon Identity
+
+_<b>Q:</b> Is this before/after example realistic/representative?_
+
+You need to handle user data synchronization manually:
```

Comment:
Here's a good example: https://github.com/clarkbw/neon-nextjs-drizzle-clerk-inngest. Look at how much complexity this feature is cutting: https://github.com/clarkbw/neon-nextjs-drizzle-clerk-inngest/blob/main/src/inngest/clerk.ts. **Not** to mention you won't have to sign up for Inngest and set up Inngest API keys and such.

Reply:
Neon Identity has 2/3 major pros  the way I see it, and I think that must be VERY clear:

1- Setup auth provider straight from the Neon dashboard
2- Neon becomes your source of truth for all data, including users and the data is always synced
3 (related to 2) - Easy Data Relationships, it becomes super simple to just use the `users_sync` table as any other table

Reply:
Agreed with Pedro, but we just have to be very careful with the expression "source of truth". With Neon Identity, at least for now, Neon will not really be the source of truth for your data but rather you'll have a sync.

Reply:
Added a benefits section:

```
## Key benefits

- Provision auth provider projects and manage your users directly from the Neon Console
- Automated synchronization of user profiles between auth provider and your database
- Easy database relationships, since your user profiles are available as any other table
```